hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -157,35 +166,32 @@ public Collection<String> getTaskTables() {
   @Override
   public Pair<String, Boolean> process(OMUpdateEventBatch events) {
     Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
-    // Initialize maps to store count and size information
-    HashMap<String, Long> objectCountMap = initializeCountMap();
-    HashMap<String, Long> unReplicatedSizeMap = initializeSizeMap(false);
-    HashMap<String, Long> replicatedSizeMap = initializeSizeMap(true);
-    final Collection<String> taskTables = getTaskTables();
```

Comment:
Aren't the map initialisations and the `getTaskTables()` still necessary here?

Reply:
> Aren't the map initialisations and the `getTaskTables()` still necessary here?

Once tables and maps are initialised once in `reprocess`, they will be available in each delta OM sync. Do you have any other use case in mind ?

Reply:
Ideally no. But why did these initialisations were there in each `process` call? I'm fine with this change, just want to make sure we are not missing out on scenarios or not changing the behaviour.

Reply:
As far as I understand, we call `process` only for delta OM DB updates to process and delta means , some DB updates have already been processed before by this background task, do repopulating these maps and collection from recon OM DB snapshot first time makes sense during `re-process` and not every time with `process`. so I don't see the reason to do initialisations of these maps or collection in every call to process because these maps are used in process to compute the value in memory and then write to DB. I am not sure why earlier these initialisations were there in each process call? @ArafatKhan2198 could know about this if he did these initialisations every time in `process` because of some other use case in his mind.

Reply:
process() can be called in normal startup where reprocess() is not called, please check init of all maps.
Additionally, taskTables() need not be kept as do not have any performance advantage as its already part of map in memory.,

Reply:
@sumitagrawl  thanks for pointing it out. This edge  case for startup of Recon in existing DB case has been taken care now.
Though taskTables()  have data already initialized and kept in memory, but we are always returning with new ArrayList and it is short lived object and creating som many objects of array list in frequent process call when high TPS should be avoided.